crates/test-utils/src/deploy_helpers.rs

```diff
@@ -831,6 +746,71 @@ impl KakarotTestEnvironmentContext {
         Self { sequencer, kakarot_client, kakarot, kakarot_contract, evm_contracts }
     }
 
+    /// Initializes the Kakarot test environment and deploys the Kakarot system and EVM contracts.
+    async fn initialize_without_dumped_state(
+        sequencer: TestSequencer,
+        starknet_provider: Arc<JsonRpcClient<HttpTransport>>,
+    ) -> Self {
+        let starknet_account = sequencer.account();
+
+        // Define the expected funded amount for the Kakarot system
+        let expected_funded_amount = FieldElement::from_dec_str("1000000000000000000").unwrap();
+
+        // Deploy the Kakarot system
+        let kakarot = deploy_kakarot_system(&sequencer, EOA_WALLET.clone(), expected_funded_amount).await;
+
+        let starknet_deployer_account = deploy_deployer_account(starknet_provider.clone(), &starknet_account).await;
+
+        // Create a Kakarot client
+        let kakarot_client = KakarotClient::new(
+            StarknetClientConfig::new(
+                Network::JsonRpcProvider(sequencer.url()),
+                kakarot.kakarot_address,
+                kakarot.proxy_class_hash,
+            ),
+            starknet_provider,
+            starknet_deployer_account,
+        );
+
+        let kakarot_contract =
+            KakarotContract::new(kakarot_client.starknet_provider(), kakarot.kakarot_address, kakarot.proxy_class_hash);
+
+        let mut test_environment =
+            Self { sequencer, kakarot_client, kakarot, kakarot_contract, evm_contracts: HashMap::new() };
+
+        // Deploy the Counter contract
+        test_environment = test_environment
+            .deploy_evm_contract(ContractDeploymentArgs { name: "Counter".into(), constructor_args: () })
+            .await;
+        let counter = test_environment.evm_contract("Counter");
```

Comment:
it's quite odd to have a `deploy` function (here  `deploy_evm_contract`) returning a mutation of its `self`, while the actual deployed object is accessed lated on with a gettter.

It's ok to me to consider that `deploy` adds some sort of record to the object, and that we then need a getter, but not really happy with the re assignment

Reply:
Gotcha will create an issue to this

Reply:
Issue here https://github.com/kkrt-labs/kakarot-rpc/issues/566